tests/p2p_stress.py

```diff
@@ -2,46 +2,94 @@
 import p2p_test_peers
 import random
 import time
+import copy
+import threading
 
 class StressNetwork:
     speeds=[1,5,10,30,60,100,500]
-    sec=5
+    sec=10
+    maxthreads=100
+    trList=[]
 
     def maxIndex(self):
         return len(self.speeds)
 
+    def randAcctName(self):
+        s=""
+        for i in range(12):
+            s=s+random.choice("abcdefghijklmnopqrstuvwxyz12345")
+        return s
+    
+    def _transfer(self, node, acc1, acc2, amount, threadId, round):
+        memo="%d %d" % (threadId, round)
+        tr = node.transferFunds(acc1, acc2, amount, memo)
+        self.trList.append(tr)
+
     def execute(self, cmdInd, node, ta, eosio):
-        print("\n==== stress network test: %d transaction(s)/s for %d secs ====" % (self.speeds[cmdInd], self.sec))
+        print("\n==== network stress test: %d transaction(s)/s for %d secs ====" % (self.speeds[cmdInd], self.sec))
         total = self.speeds[cmdInd] * self.sec
-        acclist = []
-        for k in range(total):
-            s=""
-            for i in range(10):
-                s=s+random.choice("abcdefghijklmnopqrstuvwxyz12345")
-            acclist.append(s)
-        ta.name = s
-        transIdlist = []
-        print("==== creating %d account(s) ====" % (total))
-        delay = 1.0 / self.speeds[cmdInd]
+
+        ta.name = self.randAcctName()
+        acc1 = copy.copy(ta)
+        print("creating new account %s" % (ta.name))
+        tr = node.createAccount(ta, eosio, stakedDeposit=0, waitForTransBlock=False)
+        print("transaction id %s" % (node.getTransId(tr)))
+
+        ta.name = self.randAcctName()
+        acc2 = copy.copy(ta)
+        print("creating new account %s" % (ta.name))
+        tr = node.createAccount(ta, eosio, stakedDeposit=0, waitForTransBlock=False)
+        print("transaction id %s" % (node.getTransId(tr)))
+
+        time.sleep(1.0)
+
+        print("issue currency into %s" % (acc1.name))
+        contract="eosio"
+        action="issue"
+        data="{\"to\":\"" + acc1.name + "\",\"quantity\":\"1000000.0000 EOS\"}"
+        opts="--permission eosio@active"
+        tr=node.pushMessage(contract, action, data, opts)
+        print("transaction id %s" % (node.getTransId(tr[1])))
+
+        time.sleep(1.0)
+
+        self.trList = []
+        expBal = 0
+        nthreads=self.maxthreads
+        if nthreads > self.speeds[cmdInd]:
+            nthreads = self.speeds[cmdInd]
```

Comment:
500 threads is a possibility in the speeds table, but by this logic it will never happen.  Is that intentional?

Reply:
hi, this test case is about the tests of different transaction rates. The "maxthreads" parameter is configurable. However, it's not really efficient to have many threads with very low transaction rate. In more details, we want to control the number of threads in such a way that each thread is at least sending 1 transaction/sec. Therefore, 500 threads are only possible if speeds >= 500 transactions/sec.